### PR TITLE
refactor: remove unreachable API, fold duplicated implementations, and fill the example gaps

### PR DESCRIPTION
**Pull request by nao1215**
This branch removes two exported surfaces that nothing could reach, folds five duplicated implementations into one each, and fills the example gaps the packages had accumulated. Apart from the two removals, which are breaking and recorded as such in the changelog, nothing here changes what any existing code does.

`parser.WriteTSVRecord` and `parser.WriteTSVRecordLineEnding` are removed. They had no production caller left once every write path went through `internal/writer`, and neither sqly, the examples, README nor this package's own examples referred to them. The three behaviours their tests pinned move to `internal/writer`'s own test, which is where the code performing them lives: a double quote written as data, a record that round trips through the TSV reader, and a field holding a tab or a line break refused with nothing written. `ErrTSVSyntax`, `ErrTSVUnrepresentable`, `TSVReader` and `NewTSVReader` are unchanged, so a caller matching the sentinel on a dump or a frame sees no difference.

`ContextLogger`, `SlogContextAdapter` and `NewSlogContextAdapter` are removed for the same reason. `WithLogger` takes a `Logger`, every internal call site logs through the four methods `Logger` declares, and nothing in this module calls a context-taking method, so handing a builder a `SlogContextAdapter` produced output identical to a `SlogAdapter` with the context argument never reaching slog. A caller who built one passes `WithLogger(NewSlogAdapter(l))` instead.

Five duplicated implementations come down to one each. The ACH and Fedwire loads held the same forty lines twice, differing only in the parse in front and the source format at the end, and now hand their parsed tables to `loadParsedTablesIntoDatabase`. `dialect.topLevelComma` walked the token stream with its own depth counter to find what `topLevelCommas` already finds, and now returns the first element of that answer. `parser`'s eight compression extension constants alias `internal/codec`'s rather than repeating the literals; their doc comment spells the values out, since the aliased expression is what godoc renders and `internal/codec` is not a package a caller can follow, and a new test holds every exported extension against its literal. In `frame`, `Rename` and `RenameColumns` share the rebuild through `renamed`, and `FillNA` and `FillNAByColumn` through `filledNA`, with each keeping its own validation, which is where the pairs genuinely differ. `DropNA` and `DropNASubset` are the same shape and were already folded.

The degenerate cases where those pairs answer differently are pinned by new regression tests rather than tidied away. `Rename` compares its two names before it looks for the column, so `Rename("unknown", "unknown")` is a no-op success, while `RenameColumns({"unknown": "unknown"})` checks every old name first and reports it as not found. A note on the description this work was written from: it expected `Rename(a, a)` to report "column already exists" and `RenameColumns({a: a})` to be a no-op success. On main both are no-op successes; the real divergence is the unknown-column case above, and that is what the tests now hold.

Thirty-five runnable examples are added, every one checked by `go test` and listed in README's API example index. `dialect` had none at all and gains five along with the index section every other package already had. In the root package, `ExcelSheetsInReader`, `ExcelSheetTableNames`, `(*DBBuilder).SkippedRows` and `NewReadOnlyDB` gain one each, as do the four ACH and Fedwire export entry points, whose shape a caller has to get right and which README described only in prose. The reader-backed examples show `ErrSourceUnavailable` first, so the error a caller meets and the way past it sit next to each other; each edits one field and reads it back out of the file it wrote, printing only that field because the surrounding bytes are rewritten by the format. The five remaining `String` methods follow `MalformedRowPolicy`, which was the only enum with an example. `parser` gains `NewTSVReader`'s, `parser/ach` and `parser/wire` two each, and `frame` twelve, which completes every pair where one half had an example and the other did not.

Two godocs shrink while this is being done. `Open` and `OpenContext` each carried the same window-function query with its error handling and its scan loop, which says nothing about either function and said it twice, while `ExampleOpen` and `ExampleOpenContext` already show a call that runs. `Open` keeps what only its own godoc says and `OpenContext` says it is `Open` with a context. One test gap closes alongside: `prep`'s `getStructType` returns `ErrStructSlicePointer` from six places and its table asserted only that an error came back, so each of the five refused shapes now has to wrap the sentinel.

Verified with `make test` (93.9% of statements, unchanged from main), `make lint`, `make lint-staticcheck`, `make examples`, and `env GOOS=windows GOARCH=amd64 go build ./...`. sqly was built and its tests run against this checkout through a temporary replace directive, which was removed afterwards and is not committed. `make benchmark` before and after shows B/op identical to the byte on every benchmark; the four whose ns/op moved more than 1.2x are all on paths this branch does not touch, and re-running the widest of them, `BenchmarkCompressionReaders/None`, seven times on main alone spans 112k to 188k ns/op, which is wider than the difference being measured.

## Summary by CodeRabbit

* **Breaking Changes**
  * Removed TSV writing APIs and legacy context-aware logging adapters.
  * Applications using these APIs must migrate to supported alternatives.

* **Documentation**
  * Expanded examples for file loading, parsing, dialects, DataFrames, ACH, and Fedwire.
  * Clarified file-loading behavior and context cancellation guidance.

* **Improvements**
  * Improved consistency across ACH/Fedwire loading, compression extensions, DataFrame renaming, and missing-value filling.

* **Tests**
  * Added coverage for TSV handling, parsing examples, DataFrame edge cases, and error matching.

### CODE REVIEW

File: dialect/example_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/nao1215/filesql/dialect"
+)
+
+func ExampleTranslate() {
+	// MySQL quotes an identifier with backticks, which SQLite reads as a
+	// string, so the quoting is rewritten.
```
[the remark this answers is not in the document]
Comment (nao1215) [reply dialect/example_test.go]: Fixed in b0fc09b. You are right: SQLite accepts a grave-accent-quoted name as an identifier for MySQL compatibility, so the comment claiming it reads one as a string was wrong. It now says translation normalizes MySQL backticks to SQLite's own double-quote form.

File: example_api_test.go [hunk 2]
```diff
@@ -604,3 +606,332 @@ id,name
 	// 1,Alice
 	// 2,Carol
 }
+
+// ExampleExcelSheetsInReader is ExampleExcelSheetsInFile for a workbook that
+// arrived as bytes rather than as a file.
+func ExampleExcelSheetsInReader() {
+	path := exampleWorkbook(os.TempDir())
+	defer func() { _ = os.Remove(path) }()
```
[the remark this answers is not in the document]
Comment (nao1215) [reply example_api_test.go]: Fixed in b0fc09b, and for the two pre-existing examples that had the same pattern rather than only the new one, since leaving two of three writing to the shared path would be worse than either. All three now take a directory from exampleTempDir, the os.MkdirTemp helper the ACH examples in this PR already used, and remove that directory afterwards.

File: CHANGELOG.md [hunk 3]
```diff
@@ -75,6 +79,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
...e that defines what TSV means here, so that a caller writing tab-separated records had something that refused a value the format cannot hold rather than quoting it, TSV having no quoting to quote with. What they guard has since moved to `internal/writer`, the single encoder every write path goes through, and the two were left as a wrapper over it that nothing called: the root dump, `prep` and `frame` each reach that encoder directly, and neither sqly, the examples, README nor this package's own examples referred to them. TSV output is what the output paths produce — `DumpDatabase` with `OutputFormatTSV`, `frame.DataFrame.ToTSV` and `prep`'s own output — and each of those refuses the same values for the same reason. `parser.TSVReader`, `parser.NewTSVReader`, `parser.ErrTSVSyntax` and `parser.ErrTSVUnrepresentable` are unchanged, and the sentinel is still what a refused value carries out of a dump and a frame, so a caller matching it sees no difference. The behavior the removed functions were tested for is tested where it now lives: a double quote written as data, a record that round trips through the TSV reader, and a field holding a tab or a line break refused with nothing written.
```
[the remark this answers is not in the document]
Comment (nao1215) [reply CHANGELOG.md:84]: Fixed. Both statements were true when written, but they land in the same unreleased section, so a reader of the release saw a contradiction. The writer-consolidation entry now says the two functions were left in place delegating rather than keeping a copy, that delegating turned out to be all they did, and points at the Breaking Changes entry that removes them.
